### ISSUE TITLE
Can't enable aspire dashboard

### ISSUE THREAD

**Issue by WolfspiritM** [microsoft/azure-container-apps#1224]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [x] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
Trying to migrate an existing ACA environment to Aspire.
I can't enable the aspire dashboard via the azure portal on an existing (test) container app environment.
I get the error:

Failed to provision component 'aspire-dashboard'. Error details: Failed to create config map external-auth-config-map for DotNetComponent aspire-dashboard in k8se-telemetry namespace. There will be no re-tries..

### Steps to reproduce
1. Open azure portal
2. Click on Enable for the aspire dashboard.

**Expected behavior** [What you expected to happen.]
Aspire dashboard gets enabled

**Actual behavior** [What actually happened.]
Error appears

**Screenshots**         
<img width="167" alt="image" src="https://github.com/microsoft/azure-container-apps/assets/5904171/c9a7029e-7483-4c69-ae30-c9c1767f2bc3">

### Additional context
The container app environment is part of a custom vnet. Could this maybe be related? 
Location is: Germany West Central

**Comment by simonjj** [comment]
@WolfspiritM, thank you for reaching out. Can you please try standing up a new environment? What you are seeing is very likely due to the environment being created prior to the dashboard functionality being available.

**Comment by WolfspiritM** [comment]
> @WolfspiritM, thank you for reaching out. Can you please try standing up a new environment? What you are seeing is very likely due to the environment being created prior to the dashboard functionality being available.

Hi. Thanks for the response.
The environment was created just a few days ago without the dashboard functionality enabled as a test for migrating our real environment to aspire: "createdAt": "2024-07-08T17:33:32.8304312",
I guess that's new enough?

As a side note: Does that mean the dashboard can't be enabled with (older) environments that already exist? That means migration isn't as easy cause it means larger interruptions when moving the CNAME to a new env.

**Comment by jedjohan** [comment]
I get the same error message, tried on the envs I have in staging. They are all "old" though

**Comment by simonjj** [comment]
Thanks for the report, we've determined that there is an issue which we continue to investigate. If this happens we encourage you to disable the component and wait a few minutes to reenable it again. If this continues to happen please send your environment details and a reference to acasupport please.

**Comment by jedjohan** [comment]
Any information reg the prev question:?

"Does that mean the dashboard can't be enabled with (older) environments that already exist?"

**Comment by paul-shell** [comment]
I get this with every new one I setup, even after waiting.

**Comment by simonjj** [comment]
@jedjohan No, I've since clarified that the dashboard should be available across all environments regardless of when those were created.

@paul-shell Please share your environment details with us (env name, subscription id) via acasupport at microsoft.com, we continue to investigate this but can also help you during that process.

**Comment by fardarter** [comment]
I'm getting the same error.

**Comment by simonjj** [comment]
We've identified the issue and are working on a fix. Very likely you are experiencing this issue due to your environment being in Consumption only (check the app overview page to confirm).

**Comment by fardarter** [comment]
@simonjj app overview?

We are running in consumption, yes. Should we expect it to work for consumption tier?

Not going to ask for an ETA, but is this considered at least medium priority?

Are there any additional configuration dependencies or should it work OTB in all configurations?

**Comment by simonjj** [comment]
This is being ranked above medium priority. If you switch to a new environment it will work, the issue you're experiencing is due to your environment being a _Consumption only_ and not a _Workload Consumption profile_ environment. I know these two can be very confusing but there is a major change in architecture between environments that can enable workload profiles and those that can't. We suggest for now the workaround be to run in a newer environment that allows for workload profiles.

**Comment by fardarter** [comment]
> This is being ranked above medium priority. If you switch to a new environment it will work, the issue you're experiencing is due to your environment being a _Consumption only_ and not a _Workload Consumption profile_ environment. I know these two can be very confusing but there is a major change in architecture between environments that can enable workload profiles and those that can't. We suggest for now the workaround be to run in a newer environment that allows for workload profiles.

Thanks, I apprecaite the response. 

For total clarity, this is intended to work in consumption only as well?

**Comment by RascoApps** [comment]
So we must recreate the environment to get access to the dashboard?

is there a way to retain the public IP address? this environment was created before profiles were released.

**Comment by nbwdk** [comment]
We are doing a new implementation where container apps in a consumption profile are utilized. Would be nice to know if consumption-based profiles will be supported.

**Comment by mip1983** [comment]
Also hitting the same thing trying to deploy Aspire. I've got other consumption only environments working fine with the dashboard, difference this time is it's attached to a vnet.

**Comment by rsalus** [comment]
still broken

**Comment by AdrianoAE** [comment]
Facing same issue in Workload profile (Consumption), the env is behind a private VNET
